src/parseMachine.ts

```diff
@@ -6,6 +6,29 @@ import realmsShim from 'realms-shim';
 
 const realm = realmsShim.makeRootRealm();
 
+/**
+ * Wraps a callback in a function to prevent
+ * accessing 'this'
+ */
+const wrapCallbackToPreventThis = (callback: () => void) => () => {
+  return callback();
+};
```

Comment:
Something like this should improve the compatibility with native timers:
```suggestion
const wrapCallbackToPreventThis = (callback: (...args: any[]) => void) => (...args: any[]) => {
  return callback(...args);
};
```

`setTimeout` and `setInterval` forward arguments to the inner callback, so this works:
```js
setTimeout((a, b) => {}, 0, 'a', 'b')
```

Reply:
Mmm, good catch